test_variables.py

```diff
@@ -63,7 +66,7 @@ def test_invalid_format(testdir, file_format):
     testdir.makepyfile('def test(variables): pass')
     result = run(testdir, file_format, ['invalid'])
     assert result.ret == 1
-    if sys.version_info < (3, 5, 0):
+    if sys.version_info < (3, 5, 0) or not file_format:
```

Comment:
This actually fails for me with this change, which is what I'd expect. I actually think we should revert the changes to this test, as the intention here is to test that an invalid format results in an error. For this change, I think we should introduce two new tests: `test_unsupported_format` where we check a file that ends in anything outside of the supported extensions includes the warning in stdout, and `test_missing_extension`, where we check that the warning message is not output to the console. Both tests can use valid JSON files, which also allows us to check that we default to JSON parsing.